core/core-dotspacemacs.el

```diff
@@ -121,7 +121,7 @@ with `:' and Emacs commands are executed with `<leader> :'.")
   "If non nil `Y' is remapped to `y$'.")
 
 (defvar dotspacemacs-display-default-layout nil
-  "If non nil then the `Default' layout name is displayed in the mode-line.")
+  "Name of the default layout as a string to be displayed in the mode-line.")
```

Comment:
You can rename the variable to reflect that this is about the name of the default layout and update the documentation to mention that a nil name will not show up in the mode-line.